### PR TITLE
nmcli: conn_reload param and up/down states

### LINKED ISSUE #3752: apply nmcli configuration
### Summary

Problem: how to apply the configuration on the host after launching the "nmcli" module?
Now I do it through the nmcli up handler. maybe there is another option that I don't know about?

I expected that with the "present" state, they would be applied, since this is the internal mechanism of the NetworkManager.
Perhaps a new state should be added for this?
```
ansible --version
ansible [core 2.11.1] 
  config file = /srv/infra/itpark/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/pythonEnvs/ansible/lib/python3.9/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/user/pythonEnvs/ansible/bin/ansible
  python version = 3.9.7 (default, Oct 10 2021, 15:13:22) [GCC 11.1.0]
  jinja version = 3.0.1
  libyaml = True
```

### Issue Type

Feature Idea

### Component Name

nmcli

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### LINKED ISSUE #7152: nmcli: allow module to bring interfaces up/down
### Summary

I'm a bit surprised to notice that the nmcli does not have this feature, and there's no bug report about it.

I'd like to be able to bring interfaces up or down. So the equivalent of the shell command `nmcli c up myvpn` without having to resort to the ansible command module.

Here would be an example task:
```
- name: Turn on VPN
  nmcli:
    conn_name: myvpn
    state: up
```

The requirement would be to expand the state parameter to allow the values "up"/"down", and act upon that.

If this somehow is possible by this or another module (excluding shell/command), this should be more prominently documented.

### Issue Type

Feature Idea

### Component Name

nmcli

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### LINKED ISSUE #8704: nmcli: changing routes on an interface does not lead to active routes on a host
### Summary

When I add additional routes to an existing interface using community.general.nmcli the routes are added to the connection profile but not made active on the machine.

I can see by running `nmcli show connection <foo> | grep ipv4.routes:` that the routes are updated in the connection configuration but not made active.  If I run `ip route show` the routes from the nmcli profile are not in the system's routing table.

This seems to be due to 
https://github.com/ansible-collections/community.general/blob/main/plugins/modules/nmcli.py#L2629-L2652

where the module only runs `modify` when the interface already exists.  For a new interface the interface is brought up which makes the routes active.

I've worked around this by writing a handler I can call to run `nmcli connection up <foo>` when changes happen.

### Issue Type

Bug Report

### Component Name

community.general.nmcli

### Ansible Version

```console (paste below)
ansible [core 2.12.7]
  config file = /Users/michaelhicks/gits/joyent/spc-infra/ansible/ansible.cfg
  configured module search path = ['/Users/michaelhicks/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/michaelhicks/py-venv/lib/python3.10/site-packages/ansible
  ansible collection location = /Users/michaelhicks/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/michaelhicks/py-venv/bin/ansible
  python version = 3.10.2 (v3.10.2:a58ebcc701, Jan 13 2022, 14:50:16) [Clang 13.0.0 (clang-1300.0.29.30)]
  jinja version = 3.1.2
  libyaml = True
```

### Community.general Version

```console (paste below)
# /Users/michaelhicks/py-venv/lib/python3.10/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 4.8.3

# /Users/michaelhicks/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 5.4.0
```

### Configuration

```console (paste below)
$ DEFAULT_BECOME_METHOD(/Users/michaelhicks/gits/joyent/spc-infra/ansible/ansible.cfg) = sudo
DEFAULT_FORKS(/Users/michaelhicks/gits/joyent/spc-infra/ansible/ansible.cfg) = 20
DEFAULT_REMOTE_USER(/Users/michaelhicks/gits/joyent/spc-infra/ansible/ansible.cfg) = centos
DEFAULT_VAULT_PASSWORD_FILE(/Users/michaelhicks/gits/joyent/spc-infra/ansible/ansible.cfg) = /Users/michaelhicks/gits/joyent/spc-infra/ansible/vault.txt
INTERPRETER_PYTHON(/Users/michaelhicks/gits/joyent/spc-infra/ansible/ansible.cfg) = auto_silent
```

### OS / Environment

Rocky 9

### Steps to Reproduce

```yaml (paste below)
- name: setup static routes for region_neptune things
  community.general.nmcli:
    type: ethernet
    conn_name: "System {{item.interface }}"
    state: present
    routes4: "{{ item.routes | join(',') }}"
  loop: "{{ managed_routes.v4_routes | list }}"
  register: nmcli_routes
  notify: apply_routes
  when:
    - ansible_distribution == "Rocky"
    - managed_routes
    - managed_routes.v4_routes
```

the register and notify are so I can `nmcli up` the connection to make the routes active
```yaml
- name: nmcli apply routes
  ansible.builtin.shell: "nmcli connection up '{{ item }}'"
  loop: "{{ nmcli_routes.results | selectattr('changed', 'equalto', True) | map(attribute='conn_name') | list }}"
  when: ansible_distribution == "Rocky"
  listen: "apply_routes"
```

### Expected Results

I expect `ip route show` to contain the routes that are also in `nmcli connection show <foo> | grep 'ipv4.routes:'` 

### Actual Results

routes are in the NetworkManager connection profile but not the system's routing table.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by abakanovskii**
##### SUMMARY

Add `conn_reload` option to perform `nmcli connection reload` if set to true
Add `up` and `down` states to perform `nmcli up`/`nmcli down` commands

Fixes #3752
Fixes #8704
Fixes #7152

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`nmcli`

##### ADDITIONAL INFORMATION

I am not completely sure how to make applying changes correctly as there is no distinct difference between 4 different variants in a related issue (see https://github.com/ansible-collections/community.general/issues/3752#issuecomment-1714459183) so if you have some ideas I would be glad to get comments from you :)

```yaml
# Changing with reloading afterwards
  - name: Change the property of a setting e.g. MTU and reload connection
    community.general.nmcli:
      conn_name: my-eth1
      mtu: 1500
      type: ethernet
      state: present
      conn_reload: true

# Simply disabling connection
  - name: Disable connection
    community.general.nmcli:
      conn_name: my-eth1
      state: down

# Reload and then enable connection
  - name: Reload and enable connection
    community.general.nmcli:
      conn_name: my-eth1
      state: up
      reload: true
```

### CODE REVIEW

**Review by russoz — COMMENTED** [review]
Hi @abakanovskii 

Thanks again for your contribution! A couple of small details there, other than that it LGTM.

File: plugins/modules/nmcli.py [hunk 1]
```diff
@@ -34,9 +34,11 @@
     state:
         description:
             - Whether the device should exist or not, taking action if the state is different from what is stated.
+            - Using O(state=present) to create connection will automatically bring connection up.
+            - Using O(state=up) and O(state=down) will not modify connection with other parameters.
```
[the remark this answers is not in the document]
Comment (abakanovskii) [reply plugins/modules/nmcli.py:38]: Not sure how to add these beacuse `version_added` is for new params rather then updated ones
Please check last commit if I did it right

[the remark this answers is not in the document]
Comment (abakanovskii) [reply plugins/modules/nmcli.py]: thanks!

**Review by felixfontein — COMMENTED** [review]
If nobody objects, I'll merge this next week.

**Comment by felixfontein** [comment]
@abakanovskii thanks for your contribution!
@russoz thanks for reviewing!

**Pull request merged**